browser/ui/browser_commands.cc

```diff
@@ -342,4 +354,53 @@ void ShowPlaylistBubble(Browser* browser) {
 }
 #endif
 
+void GroupTabsOnCurrentOrigin(Browser* browser) {
+  auto url =
+      browser->tab_strip_model()->GetActiveWebContents()->GetVisibleURL();
+  auto origin = url::Origin::Create(url);
+
+  std::vector<int> group_indices;
+  for (int index = 0; index < browser->tab_strip_model()->count(); ++index) {
+    auto* tab = browser->tab_strip_model()->GetWebContentsAt(index);
+    auto tab_origin = url::Origin::Create(tab->GetVisibleURL());
+    if (origin.IsSameOriginWith(tab_origin)) {
+      group_indices.push_back(index);
+    }
+  }
+  auto group_id = browser->tab_strip_model()->AddToNewGroup(group_indices);
+  auto* group =
+      browser->tab_strip_model()->group_model()->GetTabGroup(group_id);
+
+  auto data = *group->visual_data();
+  data.SetTitle(base::UTF8ToUTF16(origin.host()));
+  group->SetVisualData(data);
+}
+
+void MoveGroupToNewWindow(Browser* browser) {
+  auto* tsm = browser->tab_strip_model();
+  auto current_group_id = tsm->GetTabGroupForTab(tsm->active_index());
+  if (!current_group_id.has_value()) {
+    return;
+  }
+
+  tsm->delegate()->MoveGroupToNewWindow(current_group_id.value());
+}
+
+void CloseDuplicateTabs(Browser* browser) {
+  auto* tsm = browser->tab_strip_model();
+  auto url = tsm->GetActiveWebContents()->GetVisibleURL();
+
+  for (int i = tsm->GetTabCount() - 1; i >= 0; --i) {
```

Comment:
Out of curiosity, is there any special reason to iterate in the reverse order?

Reply:
Aha! is it because the index could be changed when closing tabs?

Reply:
Yeah, exactly!